TwitchChatVotingProxy/VotingReceiver/YTapiV1VotingReceiver.cs

```diff
@@ -0,0 +1,276 @@
+                int index = 0;
+                while ((val >> 7) > 0)
+                {
+                    byte m = (byte)(val & 0xFF | 0x80);
+                    buf[index++] = m;
+                    val >>= 7;
+                    if (index >= bufSize)
+                    {
+                        throw new Exception("index was larger than the buffer size. This shouldn't happen");
+                    }
+                }
+                buf[index] = (byte)val;
+                return buf;
+            }
+            public static byte[] concat(byte[] one, byte[] two)
+            {
+                byte[] ret = new byte[one.Length + two.Length];
+                Array.Copy(one, ret, one.Length);
+                Array.Copy(two, 0, ret, one.Length, two.Length);
+                return ret;
+            }
+
+            public static byte[] concat(byte[][] byteArrays)
+            {
+                int accumulatedSize = 0;
+                for (int i = 0; i < byteArrays.Length; i++)
+                {
+                    accumulatedSize += byteArrays[i].Length;
+                }
+                byte[] ret = new byte[accumulatedSize];
+                int destinationIndex = 0;
+                for (int i = 0; i < byteArrays.Length; i++)
+                {
+                    Array.Copy(byteArrays[i], 0, ret, destinationIndex, byteArrays[i].Length);
+                    destinationIndex += byteArrays[i].Length;
+                }
+                return ret;
+            }
+            public static byte[] tp(long a, long b, byte[] ary)
+            {
+                byte[] res = vn((b << 3) | a);
+                byte[] ret = concat(res, ary);
+                return ret;
+            }
+
+            public static byte[] rs(long a, string ary)
+            {
+                byte[] ary2 = Encoding.Default.GetBytes(ary);
+                long len = ary2.Length;
+                return tp(2, a, concat(vn(len), ary2));
+            }
+
+            public static byte[] rs(long a, byte[] ary)
+            {
+                return tp(2, a, concat(vn(ary.Length), ary));
+            }
+
+            private byte[] nm(long a, long ary)
+            {
+                return tp(0, a, vn(ary));
+            }
+
+            private string _header()
+            {
+                var S1_3 = rs(1, videoId);
+                var S1_5 = concat(rs(1, channelId), rs(2, videoId));
+
+                var S1 = concat(rs(3, S1_3), rs(5, S1_5));
+                var S3 = rs(48687757, rs(1, videoId));
+                var header_replay = concat(new byte[][] { rs(1, S1), rs(3, S3), new byte[] { 0x20, 0x01 } });
+                var realb64 = Convert.ToBase64String(header_replay);
+                var pyb64 = realb64.Replace('/', '_');
+                pyb64 = pyb64.Replace('+', '-');
+                return pyb64;
+            }
+
+            private string build(long[] ts)
+            {
+                var b1 = new byte[] { 0x08, 0x00 };

```

Comment:
Technically, doing `byte[] b1 = { 0x08, 0x00 };` would save us the `new` keyword. But I'm not that much of a c# expert to tell what's better.
But I can tell you I've no idea what's going on, lol. I'm assuming this is just copied from somewhere?

Reply:
Should I make a second pull request after all those change or is there a way to update the current one?